test.py

```diff
@@ -648,11 +650,22 @@ def print_summary(self):
     async def run(self, options):
         async with self.suite.clusters.instance() as cluster:
             self.args.insert(0, "--host={}".format(cluster[0].host))
-            cluster[0].take_log_savepoint()
-            self.success = await run_test(self, options)
-            if not self.success:
+            try:
+                cluster.pre_check()
+                self.is_pre_check_ok = True
+                cluster[0].take_log_savepoint()
+                code = await run_test(self, options)
+                cluster.post_check()
+                self.is_post_check_ok = True
+                self.success = code
+            except Exception as e:
                 self.server_log = cluster[0].read_log()
```

Comment:
The previous version saved `self.server_log` in case `not self.success` was true. Right now it only saves it in case of Exception. It may happen that no exception is thrown but `self.success == False` for example (unless I'm mistaken?). Don't we want to save the log in that case too?

Reply:
We always save the server log in testlog/. Here we save the relevant lines of the log to print it to stdout. Once again, the matter of having things printed to stdout is highly subjective. I spent quite a bit of time debugging things with CI and locally, and think that what I did is best. When the test itself fails, (return code is not 0) there is most likely noting wrong with the server, so we 're more interested in the test log, which we print, not server log, which will clutter the ouptut.

Reply:
(resolving without further action)